# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1326227135 18000
#      Tue Jan 10 15:25:35 2012 -0500
# Branch stable
# Node ID 53a9c916ba2f4997e2cec8912f5e531d99499517
# Parent  eb29a53834af6bf0859359b3950956f287a7fb24
doc: include tips for installing build dependencies on specific systems

* install.txi: Add info for install build deps with apt-get and yum.

diff --git a/doc/interpreter/install.txi b/doc/interpreter/install.txi
--- a/doc/interpreter/install.txi
+++ b/doc/interpreter/install.txi
@@ -59,16 +59,66 @@ steps as described in @ref{Building the 
 @node Build Dependencies
 @section Build Dependencies
 
 Octave is a fairly large program with many build dependencies.  You may
 be able to find pre-packaged versions of the dependencies distributed as
 part of your system, or you may have to build some or all of them
 yourself.
 
+@menu
+* Tips for Specific Systems::
+* Build Tools::
+* External Packages::
+@end menu
+
+@node Tips for Specific Systems
+@subsection Tips for Specific Systems
+
+The names of pre-compiled packages vary by system and do not always
+match exactly the names listed above.
+
+You will usually need the development version of an external dependency
+so that you get the libraries and header files for building software,
+not just for running already compiled programs.  These packages
+typically have names that end with the suffix @code{-dev} or @code{-devel}.
+
+On systems with @code{apt-get} (Debian, Ubuntu, etc.), you may be able
+to install most of the tools and external packages using a command
+similar to
+
+@example
+apt-get build-dep octave
+@end example
+
+@noindent
+The specific package name may be @code{octave3.2} or @code{octave3.4}.
+The set of required tools and external dependencies does not change
+frequently, so it is not important that the version match exactly, but
+you should use the most recent one available.
+
+On systems with @code{yum} (Fedora, Red Hat, etc.), you may be able to
+install most of the tools and external packages using a command similar to
+
+@example
+yum-builddep octave
+@end example
+
+@noindent
+The @code{yum-builddep} utility is part of the @code{yum-utils} package.
+
+For either type of system, the package name may include a version
+number.  The set of required tools and external dependencies does not
+change frequently, so it is not important that the version exactly match
+the version you are installing, but you should use the most recent one
+available.
+
+@node Build Tools
+@subsection Build Tools
+
 The following tools are required:
 
 @table @asis
 @item C++, C, and Fortran compilers
 The Octave sources are primarily written in C++, but some portions are
 also written in C and Fortran.  The Octave sources are intended to be
 portable.  Recent versions of the GNU compiler collection (GCC) should
 work (@url{http://gcc.gnu.org}).  If you use GCC, you should avoid
@@ -113,16 +163,19 @@ you modify Octave's @code{Makefile.am} f
 depend on.
 
 @item Libtool
 Package for building software libraries
 (@url{http://www.gnu.org/software/libtool}).  Libtool is required by
 Automake.
 @end table
 
+@node External Packages
+@subsection External Packages
+
 The following external packages are required:
 
 @table @asis
 @item BLAS
 Basic Linear Algebra Subroutine library
 (@url{http://www.netlib.org/blas}).  Accelerated BLAS libraries such as
 ATLAS (@url{http://math-atlas.sourceforge.net}) are recommeded for
 better performance.
